pss/pss.go

```diff
@@ -112,11 +112,23 @@ func (p *Pss) Register(topic trojan.Topic, hndlr Handler) {
 
 // Deliver allows unwrapping a chunk as a trojan message and calling its handler func based on its topic
 func (p *Pss) Deliver(c chunk.Chunk) {
-	m, _ := trojan.Unwrap(c)
-	h := p.getHandler(m.Topic)
-	if h != nil {
-		h(*m)
+	if isPotentialTrojan(c) {
+		m, _ := trojan.Unwrap(c) // if err occurs unwrapping, there will be no handler
+		h := p.getHandler(m.Topic)
+		if h != nil {
+			h(*m)
+		}
+	}
+}
+
+// isPotentialTrojan returns true if the given chunk could be trojan
+func isPotentialTrojan(c chunk.Chunk) bool {
+	if c.Type() != chunk.ContentAddressed { // chunk must be of the content-addressed type
```

Comment:
see above. the type should be set where the validator is called

Reply:
regardless of where it is set, are we in agreement on checking it in this `Valid` func?